qualtran/bloqs/qft/qft_school_book.py

```diff
@@ -0,0 +1,47 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from functools import cached_property
+
+import attrs
+import cirq
+from numpy.typing import NDArray
+
+from qualtran import GateWithRegisters, Signature
+from qualtran.bloqs.rotations.phase_gradient import PhaseGradientSchoolBook
+
+
+@attrs.frozen
+class QFTSchoolBook(GateWithRegisters):
+    r"""Textbook version of QFT using $\frac{N (N + 1)}{2}$ controlled rotations."""
```

Comment:
did you take "schoolbook" from somewhere? It rings false to my ear. Textbook sounds more natural. I can't imagine middle schoolers learning about QFT in their schoolbooks but I can imagine an introductory grad course teaching it out of a textbook :)

Reply:
I think I was reading https://arxiv.org/pdf/1904.07356.pdf about the same time. The paper uses "school book multiplication" terminology to describe the naive version and I just extended the terminology to QFT. 

Happy to change to `textbook` instead.

Reply:
(This is a minor point but I think it's interesting)

I think this actually supports my point that "schoolbook" implies younger students. Craig is talking about the method you learn in ~2nd grade. Time to start teaching QFT to elementary school students :)

Reply:
changed to textbook